frappe/desk/doctype/number_card/number_card.py

```diff
@@ -98,7 +101,10 @@ def get_percentage_difference(doc, filters, result):
 		return
 
 	previous_result = calculate_previous_result(doc, filters)
-	difference = (result - previous_result)/100.0
+	if previous_result == 0:
+		difference = result
+	else:
+		difference = ((result/previous_result)-1)*100.0
```

Comment:
Thank you for pointing this out!
I think this would be a more correct solution:
```suggestion
	if result == previous_result:
		return 0
	try:
		return ((result/previous_result)-1)*100.0
	except ZeroDivisionError:
		return None
```

Reply:
```
if previous_result == 0:
	return result
else:
	if result == previous_result:
		return 0
	else:
		return ((result/previous_result)-1)*100.0
```

We don't need to worry about zero division error as it won't reach the division part if previous_result is 0

Reply:
If `previous_result` is 0, it would be wrong to return `result`. Showing 0 or `NaN` in that case makes more sense!

Reply:
Done, updated the code.